server/middlewares/validate-before-scheduling.js

```diff
@@ -23,46 +23,60 @@ const validationMiddleware = async (context, next) => {
 	}
 
 	// The complete, and possibly updated, publisher action.
-	const { entityId, entitySlug, mode } = { ...publisherAction, ...params.data };
+	const { entityId, entitySlug, mode, locale: actionLocale } = {
+		...publisherAction,
+		...params.data,
+	};
 
 	// Run it only for the publish mode.
 	if (mode !== 'publish') {
 		return next();
 	}
 
-	// Fetch the draft that will be published.
+	// Determine the final locale: use the provided locale first, otherwise fall back to the draft’s locale.
 	const draft = await strapi.documents(entitySlug).findOne({
 		documentId: entityId,
 		status: 'draft',
+		locale: actionLocale,
 		populate: '*',
 	});
 
-	// Throw an error if there is no document with the given documentId.
 	if (!draft) {
-		throw new errors.NotFoundError(`No entity found with documentId ${entityId} for content type ${entitySlug}`);
+		throw new errors.NotFoundError(
+			`No draft found for ${entitySlug} with documentId ${entityId}`
```

Comment:
It would be nice to include the locale in the error message as well, as we use it now for querying.